--- origsrc/binutils-gdb/bfd/configure	2014-06-03 15:03:16.000000000 -0500
+++ src/binutils-gdb/bfd/configure	2014-06-03 17:34:45.457398300 -0500
@@ -15096,8 +15096,8 @@ if test "$enable_shared" = "yes"; then
 # More hacks to build DLLs on Windows.
   case "${host}" in
   *-*-cygwin*)
-    SHARED_LDFLAGS="-no-undefined"
-    SHARED_LIBADD="-L`pwd`/../libiberty -liberty -L`pwd`/../intl -lintl -lcygwin -lkernel32"
+    SHARED_LDFLAGS="-no-undefined -Wl,--exclude-libs=libiberty.a"
+    SHARED_LIBADD="-Wl,`pwd`/../libiberty/libiberty.a -L`pwd`/../intl -lintl -lcygwin -lkernel32"
   ;;
 
   # Hack to build or1k-src on OSX
--- origsrc/binutils-gdb/bfd/configure.in	2014-06-03 15:03:16.000000000 -0500
+++ src/binutils-gdb/bfd/configure.in	2014-06-03 17:34:45.481901400 -0500
@@ -575,8 +575,8 @@ changequote([,])dnl
 # More hacks to build DLLs on Windows.
   case "${host}" in
   *-*-cygwin*)
-    SHARED_LDFLAGS="-no-undefined"
-    SHARED_LIBADD="-L`pwd`/../libiberty -liberty -L`pwd`/../intl -lintl -lcygwin -lkernel32"
+    SHARED_LDFLAGS="-no-undefined -Wl,--exclude-libs=libiberty.a"
+    SHARED_LIBADD="-Wl,`pwd`/../libiberty/libiberty.a -L`pwd`/../intl -lintl -lcygwin -lkernel32"
   ;;
 
   # Hack to build or1k-src on OSX
--- origsrc/binutils-gdb/opcodes/configure	2014-06-03 15:03:51.000000000 -0500
+++ src/binutils-gdb/opcodes/configure	2014-06-03 17:34:45.492402700 -0500
@@ -12455,8 +12455,8 @@ if test "$enable_shared" = "yes"; then
 
   case "${host}" in
     *-*-cygwin*)
-      SHARED_LDFLAGS="-no-undefined"
-      SHARED_LIBADD="-L`pwd`/../bfd -lbfd -L`pwd`/../libiberty -liberty -L`pwd`/../intl -lintl -lcygwin"
+      SHARED_LDFLAGS="-no-undefined -Wl,--exclude-libs=libiberty.a"
+      SHARED_LIBADD="`pwd`/../bfd/libbfd.la -Wl,`pwd`/../libiberty/libiberty.a -L`pwd`/../intl -lintl -lcygwin"
       ;;
    *-*-darwin*)
      SHARED_LIBADD="-Wl,`pwd`/../bfd/.libs/libbfd.dylib ${SHARED_LIBADD}"
--- origsrc/binutils-gdb/opcodes/configure.in	2014-06-03 15:03:51.000000000 -0500
+++ src/binutils-gdb/opcodes/configure.in	2014-06-03 17:34:45.513405400 -0500
@@ -178,8 +178,8 @@ changequote([,])dnl
 
   case "${host}" in
     *-*-cygwin*)
-      SHARED_LDFLAGS="-no-undefined"
-      SHARED_LIBADD="-L`pwd`/../bfd -lbfd -L`pwd`/../libiberty -liberty -L`pwd`/../intl -lintl -lcygwin"
+      SHARED_LDFLAGS="-no-undefined -Wl,--exclude-libs=libiberty.a"
+      SHARED_LIBADD="`pwd`/../bfd/libbfd.la -Wl,`pwd`/../libiberty/libiberty.a -L`pwd`/../intl -lintl -lcygwin"
       ;;
    *-*-darwin*)
      SHARED_LIBADD="-Wl,`pwd`/../bfd/.libs/libbfd.dylib ${SHARED_LIBADD}"
